MDANSE/Src/MDANSE/Framework/Jobs/MolecularTrace.py

```diff
@@ -154,13 +153,13 @@ def run_step(self, index):
 
         grid = np.zeros(self.gdim, dtype=np.int32)
 
-        # Loop over the indexes of the selected atoms for the molecular trace calculation.
-        mt_fast_calc.mt(
-            conf["coordinates"][self._indexes, :],
-            grid,
-            self.configuration["spatial_resolution"]["value"],
-            self.min,
-        )
+        resolution = self.configuration["spatial_resolution"]["value"]
+
+        indices = np.floor(
+            (conf["coordinates"][self._indexes, :] - self.min.reshape((1, 3)))
+            / resolution
+        ).astype(int)
+        grid[tuple(indices.T)] += 1
```

Comment:
I think there is a slight difference to the cython code when there are a duplicated set of indices  e.g.

```
a = np.zeros((2, 2))
a
Out[21]: 
array([[0., 0.],
       [0., 0.]])
a[(np.array([0, 1, 1]), np.array([0, 1, 1]))] += 1
a
Out[23]: 
array([[1., 0.],
       [0., 1.]])
```

I ran with a DFTB water trajectory and there are some small differences in the output.


OLD
![image](https://github.com/user-attachments/assets/be2f474a-41b6-40c7-a3c4-ba9eacb74ed4)

NEW
![image](https://github.com/user-attachments/assets/584cb09e-e4dd-4afd-87af-760ef94af540)